skiffWindowsApp/Skiff Desktop/TrayController.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Drawing;
+using System.Windows;
+using System.Windows.Forms;
+
+namespace Skiff_Desktop
+{
+    internal class TrayController
+    {
+        private NotifyIcon _trayIcon;
+        private ToolStripMenuItem _traySettingMenuItem;
+        private Icon _iconNormal;
+        private Icon _iconUnreadDot;
+        private MainWindow _mainWindow;
+
+
+        public TrayController(MainWindow mainWindow)
+        {
+            _mainWindow = mainWindow;
+            _mainWindow.StateChanged += OnWindowStateChanged;
+            _mainWindow.UnreadCounterChanged += UpdateTrayInfo;
+            _mainWindow.Closed += OnWindowClosed;
+
+            InitIconsBitmaps();
+            InitTrayIcon();
+            UpdateTrayInfo();
+        }
+
+        private void InitIconsBitmaps()
+        {
+            IntPtr iconHandle = Properties.Resources.logo.GetHicon();
+            _iconNormal = Icon.FromHandle(iconHandle);
+
+            Color niceRed = Color.FromArgb(255, 240, 71, 71);
+            Brush bgColor = new SolidBrush(niceRed);
+            Bitmap bitmap = Properties.Resources.logo; // bitmap.Width: 32
+            Graphics graphics = Graphics.FromImage(bitmap);
+            int diameter = 14;
+            int coord = 18;
+            Rectangle dotPosAndSize = new Rectangle(coord, coord, diameter, diameter);
+            graphics.FillEllipse(bgColor, dotPosAndSize);
+            // If at some point we want to render the counter, we will need this, but will be redraw each time, so wont go here.
+            //graphics.DrawString(_count.ToString(), new Font("Arial Unicode MS", 11), Brushes.White, new PointF(coord, coord));
+            _iconUnreadDot = Icon.FromHandle(bitmap.GetHicon());
```

Comment:
`iconHandle` needs to be passed into `DestroyIcon`
`_iconNormal`should be disposed at some point
`graphics` needs to be disposed
`bitmap` needs to be disposed
`_iconUnreadDot` should be disposed at some point

Consider perhaps using a [SafeHandle](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.safehandle?view=net-6.0) implementation to handle icon lifetime for you, or explicitly implement a Dispose pattern.